cpp/src/aztec/honk/composer/composer_helper/standard_plonk_composer_helper.hpp

```diff
@@ -0,0 +1,197 @@
+    std::shared_ptr<bonk::ReferenceStringFactory> crs_factory_;
+
+    std::vector<uint32_t> recursive_proof_public_input_indices;
+    bool contains_recursive_proof = false;
+    bool computed_witness = false;
+
+    StandardPlonkComposerHelper()
+        : StandardPlonkComposerHelper(
+              std::shared_ptr<bonk::ReferenceStringFactory>(new bonk::FileReferenceStringFactory("../srs_db/ignition")))
+    {}
+    StandardPlonkComposerHelper(std::shared_ptr<bonk::ReferenceStringFactory> crs_factory)
+        : crs_factory_(std::move(crs_factory))
+    {}
+
+    StandardPlonkComposerHelper(std::unique_ptr<bonk::ReferenceStringFactory>&& crs_factory)
+        : crs_factory_(std::move(crs_factory))
+    {}
+    StandardPlonkComposerHelper(std::shared_ptr<bonk::proving_key> p_key, std::shared_ptr<bonk::verification_key> v_key)
+        : circuit_proving_key(std::move(p_key))
+        , circuit_verification_key(std::move(v_key))
+    {}
+    StandardPlonkComposerHelper(StandardPlonkComposerHelper&& other) noexcept = default;
+    StandardPlonkComposerHelper(const StandardPlonkComposerHelper& other) = delete;
+    StandardPlonkComposerHelper& operator=(StandardPlonkComposerHelper&& other) noexcept = default;
+    StandardPlonkComposerHelper& operator=(const StandardPlonkComposerHelper& other) = delete;
+    ~StandardPlonkComposerHelper() = default;
+
+    inline std::vector<bonk::SelectorProperties> standard_selector_properties()
+    {
+        std::vector<SelectorProperties> result{
+            { "q_m", false }, { "q_c", false }, { "q_1", false }, { "q_2", false }, { "q_3", false },
+        };
+        return result;
+    }
+    void add_recursive_proof(CircuitConstructor& circuit_constructor,
+                             const std::vector<uint32_t>& proof_output_witness_indices)
+    {
+
+        if (contains_recursive_proof) {
+            circuit_constructor.failure("added recursive proof when one already exists");
+        }
+        contains_recursive_proof = true;
+
+        for (const auto& idx : proof_output_witness_indices) {
+            circuit_constructor.set_public_input(idx);
+            recursive_proof_public_input_indices.push_back((uint32_t)(circuit_constructor.public_inputs.size() - 1));
+        }
+    }
+    std::shared_ptr<bonk::proving_key> compute_proving_key(const CircuitConstructor& circuit_constructor);
+    std::shared_ptr<bonk::verification_key> compute_verification_key(const CircuitConstructor& circuit_constructor);
+
+    void compute_witness(const CircuitConstructor& circuit_constructor) { compute_witness_base(circuit_constructor); }
+
+    plonk::Verifier create_verifier(const CircuitConstructor& circuit_constructor);
+    /**
+     * Preprocess the circuit. Delegates to create_prover.
+     *
+     * @return A new initialized prover.
+     */
+    plonk::Prover preprocess(const CircuitConstructor& circuit_constructor)
```

Comment:
Cody removed this `preprocess` function in the linearization trick removal PR. You might be able to just delete this since all of the tests should have been updated to avoid using it.

Reply:
Deleted